### PR TITLE
GLX: Define the GLX dispatch stubs at compile time (v2)

### PR DESCRIPTION
**Pull request by kbrenneman**
This is a rebase of #129. Besides the rebase, I rewrote the aarch64 stubs to fix a crash that shows up when you turn on compiler optimizations.

With this change, the dispatch stubs for GLX functions are defined at compile time instead of allocating executable memory to generate them at runtime.

This requires rewriting the stubs so that they look up the functions from a global array instead of patching the addresses into the stubs themselves. Unlike libGLdispatch, though, it can use a single, static array instead of having to look up a per-thread array from another DSO.

### CODE REVIEW

File: src/GLX/glvnd_genentry.c [hunk 1]
```diff
@@ -49,299 +49,157 @@
...
-    0xd503201f,
-    // Offset that needs to be patched
-    // 1:
-    0x00000000, 0x00000000,
-};
-
-static const int DISPATCH_FUNC_OFFSET = 12;
+#define STUB_SIZE 16
+#define STUB_ASM_ARCH(slot) \
+    "adrp x16, entrypointFunctions + " slot "*8\n" \
```
Comment (inolen) [inline src/GLX/glvnd_genentry.c]: Does the assembler convert this into:
```adrp x16, label
add x16, slot * 8```

I ask as I believe `adrp` will place the page-aligned address into `xd`. On that note, I think entrypointFunctions would also needs to be guaranteed to be 4k aligned.

Comment (kbrenneman) [reply src/GLX/glvnd_genentry.c]: It doesn't generate a separate `add` instruction, but as far as I understand, it's not needed.

The `adrp` instruction will load all but the low-order 12 bits of the address into x16, and then the `ldr` instruction takes the remaining offset as an immediate value.

For both instructions, it's loading the address `(entrypointFunctions + slot * 8)`, not just `(entrypointFunctions)` by itself.

Comment (inolen) [reply src/GLX/glvnd_genentry.c]: Ah, I wasn't paying attention to the offset passed to ldr. Looks good then 👍

**Comment by kbrenneman** [comment]
We've migrated the repository over to GitLab, so this is now tracked at https://gitlab.freedesktop.org/glvnd/libglvnd/merge_requests/146.

**Pull request closed without merging**